packages/typewiz-core/src/configuration-parser.ts

```diff
@@ -0,0 +1,65 @@
+import * as Ajv from 'ajv';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+import * as util from 'util';
+import { IApplyTypesOptions } from './apply-types';
+import { ICompilerOptions } from './compiler-helper';
+import { IInstrumentOptions } from './instrument';
+const readFileAsync = util.promisify(fs.readFile);
+
+export class ConfigurationParser {
+    private typewizConfig: any;
+    private configurationPath: string | undefined;
+
+    public findConfigFile(cwd: string): string {
+        return ts.findConfigFile(cwd, ts.sys.fileExists, 'typewiz.json');
+    }
+
+    public async parse(configurationPath: string): Promise<void> {
+        const typewizConfigSchema = JSON.parse(
+            await readFileAsync(path.join(__dirname, 'typewiz.json'), { encoding: 'utf8' }),
+        );
+
+        let typewizConfigString;
+        try {
+            typewizConfigString = await readFileAsync(path.resolve(configurationPath), { encoding: 'utf8' });
```

Comment:
How about creating an internal `parseConfig()` function that will take the config as an object? then we could provide `parse()` and `parseSync()` as wrappers, and use `parseSync()` in `typewiz-core/register`, so we keep the node registration API sync and simple.

Reply:
Which file do you mean with `typewiz-core/register` (I moved `typewiz-core/node-register` to `typewiz-node/node-register` since that is the place it is used). In general the way the configuration parser is currently used in `typewiz-webpack` and `typewiz-node` it would only make sense to provide a synchronous alternative if it was used for everything (which @MadaraUchiha  explicitly didn't want in the first review).

Reply:
My bad - I meant the `register()` function exported by `typewiz-node`. The reason we need that one to be sync is for using with mocha. You use it like:

`mocha -r typewiz-node/dist/register src/**/test.ts`

If this would be done asynchronously, there is a race condition where mocha might start requiring .ts files before we have registered.

Reply:
Makes sense. But that does mean we have to use the synchronous parsing everywhere in typewiz-node (index.ts and node-register.ts).

Reply:
That sounds okay to me. `ts-node` and `babel` both have [synchronous `register` methods](https://babeljs.io/docs/usage/babel-register/), and as far as I can tell, they also read configuration files.

Reply:
Done